.gemini/styleguide.md

```diff
@@ -0,0 +1,715 @@
+    is based on `keras_hub.tokenizers.WordPieceTokenizer`.
+    
+    Args:
+        vocabulary: list of strings or str. A list of strings or a string filename path.
+        lowercase: bool. If `True`, the input text will be first lowered before tokenization.
+        
+    Examples:
+    ```python
+    # Unbatched input.
+    tokenizer = keras_hub.models.MyModelTokenizer.from_preset("my_model_base")
+    tokenizer("The quick brown fox jumped.")
+    ```
+    """
+    
+    backbone_cls = MyModelBackbone
+    
+    def __init__(
+        self,
+        vocabulary=None,
+        lowercase=False,
+        **kwargs,
+    ):
+        super().__init__(vocabulary=vocabulary, lowercase=lowercase, **kwargs)
+        
+        # Add special tokens
+        self.start_token_id = self.token_to_id("[CLS]")
+        self.end_token_id = self.token_to_id("[SEP]")
+        self.pad_token_id = self.token_to_id("[PAD]")
+        self.mask_token_id = self.token_to_id("[MASK]")
+```
+
+### Preprocessors (`<model_name>_preprocessor.py`)
+
+**Purpose**: A preprocessor is a `keras.layers.Layer` that orchestrates the entire preprocessing pipeline, turning raw user input into model-ready tensors.
+
+**Structure**: It internally instantiates and uses the model's specific Converter (e.g., `MyModelTokenizer`).
+
+**Functionality**: It handles padding, truncation, generating attention masks, and formatting the output into a dictionary of tensors that match the backbone's input signature (e.g., `{"token_ids": ..., "padding_mask": ...}`).
+
+**Example Preprocessor**:
+```python
+@keras_hub_export("keras_hub.models.MyModelPreprocessor")
+class MyModelPreprocessor(TextClassifierPreprocessor):
+    """MyModel preprocessing for text classification.
+    
+    This preprocessing layer will prepare inputs for text classification.
+    
+    Args:
+        tokenizer: `keras_hub.models.MyModelTokenizer`. A tokenizer instance.
+        sequence_length: int. The length of the packed inputs.
+        
+    Examples:
+    ```python
+    preprocessor = keras_hub.models.MyModelPreprocessor.from_preset("my_model_base")
+    preprocessor("The quick brown fox jumped.")
+    ```
+    """
+    
+    backbone_cls = MyModelBackbone
+    tokenizer_cls = MyModelTokenizer
+    
+    def __init__(
+        self,
+        tokenizer,
+        sequence_length=512,
+        **kwargs,
+    ):
+        super().__init__(tokenizer=tokenizer, sequence_length=sequence_length, **kwargs)
+```
+
+### Task Models (`<model_name>_<task>.py`)
+
+**Purpose**: A task model combines a Backbone, a Preprocessor, and a task-specific head (e.g., a classification or generation head).
+
+**Structure**: It should be a class that inherits from `keras.Model`.
+
+**API**: Provide a simple, high-level API for end-users, such as `predict()`, `fit()`, and `generate()`.
+
+**Example Task Model**:
+```python
+@keras_hub_export("keras_hub.models.MyModelTextClassifier")
+class MyModelTextClassifier(TextClassifier):
```

Comment:
Do you think it's better if we show an example of `CausalLM` here, given most of our models will be be `CausalLM`s? Not necessary to do this, just wondering

Reply:
done